### ISSUE TITLE
[Feature Request]: Can I use UTF-8 in compression?

### ISSUE THREAD

**Issue by wihn2021** [rikyoz/bit7z#91]
### Feature description

Is there a method that I can compress a directory using UTF-8? I have checked head files, but didn't find a method to change the encoding. 

### Code of Conduct

- [x] By submitting this issue, I agree to follow bit7z's [Code of Conduct](https://github.com/rikyoz/bit7z/blob/master/CODE_OF_CONDUCT.md)

**Comment by rikyoz** [comment]
Hi!
> Is there a method that I can compress a directory using UTF-8? I have checked head files, but didn't find a method to change the encoding.

Do you mean the encoding of the strings passed to bit7z?
In this case, there's no way in the current stable version.

Natively, Windows uses the UTF-16 encoding for Unicode strings, so you usually need to use `std::wstring` or `wchar_t` C strings. This is the approach of 7-zip, and hence of bit7z.

This choice will change from the next major version of bit7z (v4.0): since it will become a cross-platform library, by default, it will follow the so-called [UTF-8 Everywhere](https://utf8everywhere.org/) manifesto. Hence, all string parameters will be _considered_ as UTF-8 encoded `std::string`s.

**Comment by wihn2021** [comment]
I mean the directory names and filenames in zip archive. I want them to be encoded in UTF-8, so that I can use the archive on Android platform. If I use 7z.exe, I would add the argument 'cu=on' to get a zip archive with filenames encoded in UTF-8, so is there a way to use bit7z functions to get the same result as 7z argument 'cu=on'?

**Comment by rikyoz** [comment]
Uhm, I see! I actually didn't know this specific option for the zip archives.
So no, I'm sorry, there's no way in bit7z since it doesn't specify the `cu` parameter when calling the 7-zip API.
According to the [help documentation](https://sevenzip.osdn.jp/chm/cmdline/switches/method.htm) of 7-zip:
`By default (if cl and cu switches are not specified), 7-Zip uses UTF-8 encoding only for file names that contain symbols unsupported by local code page.`
I don't know if this can help you.
I'll need to evaluate how to add this feature to the library and whether to do it on the current stable version or add it to the next version.

**Comment by wihn2021** [comment]
Thank you for your reply!

**Comment by rikyoz** [comment]
Implemented in [v4.0.0](https://github.com/rikyoz/bit7z/releases/tag/v4.0.1).

Now, you can use 7-zip format settings like `cu` using the `setFormatProperty` method.
E.g.:
```cpp
BitFileCompressor compressor{ lib, BitFormat::Zip };
compressor.setFormatProperty( L"cu", true );
compressor.compressDirectory( "dir/path/", "dir_archive.zip" );
```

Please note that the method only accepts wide-character literals as the first argument.
